dags/financial_data_dag.py

```diff
@@ -0,0 +1,132 @@
+    soup = BeautifulSoup(r_html, 'html.parser')
+    components_table = soup.find_all(id="constituents")
+
+    # the first section (index=0) in the components table contains
+    # the headers (which are unimportant in this context) therefore,
+    # only data from index=1 on is taken. moreover, each row in the
+    # table is stored in a <tr> division, so we find all of those.
+    # The data is stored in an array, where each element contains
+    # information about a S&P 500 company.
+    data_rows = components_table[0].find("tbody").find_all("tr")[1:]
+    tickers = []
+
+    # extracting the tickers from the data
+    for row in range(len(data_rows)):
+        stock = list(filter(None,data_rows[row].text.split("\n")))
+        symbol = stock[0]
+
+        if symbol.find('.') != -1:
+            symbol = symbol.replace('.', '-')
+
+        tickers.append(symbol)
+
+    tickers.sort()
+
+    return tickers
+
+def download_yfinance_data_function(start_date):
+    """downloads Adjusted Close data from SP500 companies"""
+
+    tickers = get_sp500_ticker_symbols()
+    data = yf.download(tickers, start=start_date)['Adj Close']
+    return data.to_json()
+
+def prepare_data_function(ti):
+    """creates an array of formatted and clean data values"""
+
+    # pulling data (as string)
+    string_data = ti.xcom_pull(task_ids='download_data_task')
+
+    # transforming to json
+    json_data = json.loads(string_data)
+
+    # transforming to dataframe for easier manipulation
+    df = pd.DataFrame.from_dict(json_data, orient='index')
+
+    values_array = []
+
+    for col in range(len(df.columns)):
+        closing_date = df.columns[col]
+
+        for row in range(len(df.index)):
+            ticker = df.index[row]
+            adj_close = df.iloc[row, col]
+
+            if not(adj_close is None or math.isnan(adj_close)):
+                values_array.append(f"({closing_date}, '{ticker}', {adj_close}")
+                #values_array.append("({},\'{}\',{})".format(closing_date, ticker, adj_close))
+
+    return values_array
+
+def insert_data_function(ti):
+    """inserts financial data values into CrateDB"""
+
+    values_array = ti.xcom_pull(task_ids='prepare_data_task')
+
+    insert_stmt = "INSERT INTO sp500 (closing_date, ticker, adjusted_close) VALUES "
+    insert_stmt += ", ".join(values_array) + ";"
+
+    insert_data_task = PostgresOperator(
+                task_id="insert_data_task",
+                postgres_conn_id="cratedb_connection",
+                sql=insert_stmt
+                )
+
+    insert_data_task.execute(dict())
+
+
+with DAG(
+    dag_id="financial_data_dag",
+    start_date=datetime.datetime(2022, 1, 10),
+    schedule_interval="@daily",
+    catchup=False,
+) as dag:
+
+    download_data_task = PythonOperator(task_id='download_data_task',
+                                    python_callable=download_yfinance_data_function,
+                                    provide_context=True,
```

Comment:
I noticed this produces a deprecation warning during the pytest run:
```
tests/dag_import_test.py::test_no_import_errors
  dags/financial_data_dag.py:112: DeprecationWarning: provide_context is deprecated as of 2.0 and is no longer required
    download_data_task = PythonOperator(task_id='download_data_task',
```
Looks like `provide_context` can just be removed on all tasks.